src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -80,21 +80,29 @@ class Docker implements Serializable {
             def parsedArgs = args.split(/ (?=([^"']*["'][^"']*["'])*[^"']*$)/)
             def dir = parsedArgs[-1] ?: '.'
 
-            // Detect custom Dockerfile:
+            // iterate through arguments to check for extra variables that need to be set
+            def target = null
             def dockerfile = "${dir}/Dockerfile"
             for (int i=0; i<parsedArgs.length; i++) {
                 def arg = parsedArgs[i]
-                if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
+                // determine build target
+                if (arg.startsWith('--target=')) {
+                    target = arg.split('=')[1]
+                }
+                else if (arg.startsWith('--target') && i < (parsedArgs.length - 1)) {
+                    target = parsedArgs[i+1]
+                }
+                // determine custom dockerfile
+                else if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
```

Comment:
fwiw, this case is wrong for:
`['--file=foo']`
parsedArgs=['--file=foo']
i=0
arg = '--file=foo'
0 < 1 - 1